hw/ip/mbx/doc/programmers_guide.md

```diff
@@ -0,0 +1,60 @@
+An additional constraint of at most 1K DWORDs (4KiB) applies to the size of any response from the RoT (Outbox message).
+
+## Initialization
+
+Since the mailbox logic acts as a host on the TileLink fabric, before requests may be received from the SoC side the firmware of the RoT must configure the address ranges that may be accessed by the Inbox (SoC->RoT) and Outbox (RoT->SoC) logic.
+
+To configure each of the Inbox and Outbox sides two registers must be programmed with a suitable address range:
+
+1. Program [`inbound_base_address`](registers.md#inbound_base_address) with the _inclusive_ base address of the Inbox memory region.
+2. Program [`inbound_limit_address`](registers.md#inbound_limit_address) with the _inclusive_ end address of the Inbox, i.e. this is the highest valid address to which a DWORD may be written by the Inbox logic.
+3. Program [`outbound_base_address`](registers.md#outbound_base_address) with the _inclusive_ base address of the Outbox memory region.
+4. Program [`outbound_limit_address`](registers.md#outbound_limit_address) with the _inclusive_ end address of the Outbox, i.e. this is the highest valid address from which a DWORD may be read by the Outbox logic.
+
+Having supplied the address ranges, the RoT firmware shall declare the ranges valid by writing a value of 1 to the [`address_range_valid`](registers.md#address_range_valid) register, and may optionally lock the address range by writing MuBi4False to register [`address_range_regwen`](registers.md#address_range_regwen).
+Clearing the 'write enable' register in this manner prevents any further changes to the address range configuration until the mailbox IP block is reset, protecting against accidental or malicious changes.
+
+## Request message from SoC to RoT
+
+A software/hardware agent on the SoC side is required to check that the BUSY bit is clear before starting to send a request to the RoT.
+The BUSY bit indicates that the mailbox is already part-way through a request-response operation, so another request cannot be accepted.
+
+The request itself is sent by writing each 32-bit DWORD of the message to the WDATA register on the SoC-side interface.
+The write data will be accepted into the Inbox handler when it can be presented onto the RoT-side fabric for writing to the Inbox memory.
+A back-pressure/flow control mechanism prevents data loss at this point in the event that the SoC side faster.
+
+The request is written into the mailbox memory of the RoT side starting at the address specified in [`inbound_base_address`](registers.md#inbound_base_address) and incrementing for each DWORD transferred.
+When the entire request has been written to the Inbox via the WDATA register, the SoC-side agent must set the GO bit of the [`soc_control`](registers.md#soc_control) register to present it to the RoT firmware.
+The causes the Inbox handler to assert the `mbx_ready` to the RoT firmware so that it may collect and process the request message.
```

Comment:
```suggestion
This causes the Inbox handler to assert the `mbx_ready` to the RoT firmware so that the firmware may collect and process the request message.
```